app/helpers/service_discovery_helper.rb

```diff
@@ -5,7 +5,7 @@ def parse_request
 
 def validates_presence_of(*args)
   if args[-1].instance_of? Hash
-    map = args[-1][:in] || @input
+    map = args[-1][:in] || args[-1] || @input
```

Comment:
Why do you want to keep it as args[-1]? if you can add a key :in, the validation would become much more readable.